### PR TITLE
Login networking

### PR DESCRIPTION
**Pull request by guvanch-amangeldiyev**

### CODE REVIEW

**Review by ilya-safyan-idf — APPROVED** [review]

File: packages/data/lib/dio/token_interceptor.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    return super.onError(err, handler);
+  }
+
+  @override
+  void onResponse(Response response, ResponseInterceptorHandler handler) {
+    //TODO impl response
+
+    super.onResponse(response, handler);
+  }
+}
```
Comment (guvanch-amangeldiyev) [inline packages/data/lib/dio/token_interceptor.dart:30]: description of response about implementing

File: packages/data/lib/di/data_injector.dart [hunk 2]
```diff
@@ -13,10 +17,36 @@ Future<void> initDataModule() async {
...
+    sl.get<LogInterceptor>(),
+    sl.get<TokenInterceptor>(),
+  ];
+
   sl.registerSingleton<Dio>(
     dioBuilder(
       ApiHelperCore.baseUrl,
-      headers: ApiHelperCore.authHeader,
+      listOfInterseptors,
+      //ApiHelperCore.authHeader,
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/lib/di/data_injector.dart]: delete it

File: packages/data/lib/dio/dio_builder.dart [hunk 3]
```diff
@@ -1,15 +1,11 @@
...
-  Map<String, dynamic>? headers,
-}) {
+Dio dioBuilder(String baseUrl, List<Interceptor> interceptors,
+    {Map<String, dynamic>? headers}) {
   const timeout = 60000;
   final headersMap = headers ?? <String, dynamic>{};
-  headersMap[HttpHeaders.userAgentHeader] =
-      '${Platform.operatingSystem}-APPLICATION'.toUpperCase();
+  // headersMap[HttpHeaders.userAgentHeader] =
+  //     '${Platform.operatingSystem}-APPLICATION'.toUpperCase();
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/lib/dio/dio_builder.dart]: and it

File: packages/data/lib/dio/token_interceptor.dart (hunk 1 above)
Comment (sergei-mikhailovskii-idf) [inline packages/data/lib/dio/token_interceptor.dart]: why do u need to override onResponse?

File: packages/data/lib/repository/network_repository.dart [hunk 4]
```diff
@@ -18,24 +17,32 @@ class NetworkRepository extends ApiBaseRepositoryImpl
     this._cancelToken,
   ) : super(cancelToken: _cancelToken);
 
+//! getJobs
   @override
-  ApiAuthorizationResponse? getdata;
+  Future<ApiAuthorizationResponse?> getJobs(String url) {
+    return _service
+        .get(path: url + ApiHelperCore.pathToken)
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/lib/repository/network_repository.dart]: baseurl should be specified in dio. here u should pass only path

File: packages/data/test/networking_repository_mocks.mocks_test.dart [hunk 5]
```diff
@@ -0,0 +1,147 @@
+// Mocks generated by Mockito 5.1.0 from annotations
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/test/networking_repository_mocks.mocks_test.dart]: ignore

File: packages/domain/test/domain_repo_test.dart [hunk 6]
```diff
@@ -0,0 +1,27 @@
+// import 'package:domain/model/api_auth_response.dart';
```
Comment (sergei-mikhailovskii-idf) [inline packages/domain/test/domain_repo_test.dart]: delete

File: packages/domain/test/domain_repo_test.mocks.dart [hunk 7]
```diff
@@ -0,0 +1,58 @@
+// // Mocks generated by Mockito 5.1.0 from annotations
```
Comment (sergei-mikhailovskii-idf) [inline packages/domain/test/domain_repo_test.mocks.dart]: delete

File: packages/presentation/lib/di/presentation_injector.dart [hunk 8]
```diff
@@ -41,4 +61,20 @@ void initPresentationModule() {
...
+      sl.get<AnalyticsService>(),
+      sl.get<TokenUseCase>(),
+    ),
+  );
+
+  sl.registerFactory<MainBloc>(() => MainBloc(
+      sl.get<AnalyticsEvent>(),
+      sl.get<AnalyticsService>(),
+      sl.get<GetViewsUseCase>(),
+      sl.get<GetPrimaryViewUseCase>()));
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/di/presentation_injector.dart]: comma

File: packages/presentation/lib/internal/service/analitic_event.dart [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    //as event names may only contain alphanumeric characters and underscores
+    final characters = event.toLowerCase().split('');
+
+    for (int i = 0; i < characters.length; i++) {
+      if (characters[i] == '-') {
+        characters[i] = '_';
+      }
+    }
+
+    return characters.join();
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/internal/service/analitic_event.dart]: read about operator replaceAll

File: packages/presentation/lib/screen/app/my_app.dart [hunk 10]
```diff
@@ -50,17 +49,14 @@ class _AppState extends BlocState<App, AppBloc> {
...
       },
     );
   }
 
   Navigator _content(
-    BuildContext context,
-    BlocData? blocData,
-    AppData appData,
-  ) {
+      BuildContext context, BlocData? blocData, AppData appData) {
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/app/my_app.dart]: comma

File: packages/presentation/lib/screen/home/home_page.dart [hunk 11]
```diff
@@ -85,12 +85,16 @@ class _BuildInitialState extends StatelessWidget {
...
   Widget build(BuildContext context) {
     return ListView.separated(
         itemBuilder: (context, index) => CardItem(
-              maintitlle: screenData.jobs?[index].name ?? "empty",
-              color: screenData.jobs?[index].color ?? "empty",
+            maintitlle: screenData.jobs?[index].name ?? "empty",
+            color: GetIt.I
+                .get<ColorMapper>()
+                .getColorByName(screenData.jobs?[index].color)
+            //screenData.jobs?[index].color ?? "empty",
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/home/home_page.dart]: delete

File: packages/presentation/lib/screen/spalsh/bloc/bloc_main.dart [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+    final isAuthorized = await _getTokenUseCase();
+
+    if (isAuthorized) {
+      _analyticsService.trackCustomEvent(
+        event: AnalyticsEvent.mainScreenOpenedEvent,
+      );
+      navigateToMainPage();
+    }
+    _analyticsService.trackCustomEvent(
+        event: AnalyticsEvent.loginScreenOpenedEvent);
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/spalsh/bloc/bloc_main.dart]: comma

File: packages/presentation/lib/screen/spalsh/main_spalsh.dart [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+          child: Center(
+        child: Column(mainAxisAlignment: MainAxisAlignment.center, children: [
+          SvgPicture.asset("assets/svg/jenkis.svg"),
+          SizedBox(height: 20),
+          Text("JenkisApp",
+              style: TextStyle(
+                  fontSize: 32,
+                  fontFamily: "Georgia",
+                  color: Colors.white,
+                  fontWeight: FontWeight.w700)),
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/spalsh/main_spalsh.dart]: move styles to the separate file

File: packages/presentation/lib/screen/spalsh/main_spalsh.dart [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: AppColors.accentGreen,
+      body: SafeArea(
+          child: Center(
+        child: Column(mainAxisAlignment: MainAxisAlignment.center, children: [
+          SvgPicture.asset("assets/svg/jenkis.svg"),
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/spalsh/main_spalsh.dart]: move image paths to the separate file

File: packages/presentation/lib/widgets/card_screen.dart [hunk 15]
```diff
@@ -47,14 +46,6 @@ class CardItem extends StatelessWidget {
...
 
-  Color? stringToColor(String color) {
-    if (color.contains("red")) {
-      return Colors.red;
-    } else {
-      return Colors.blue;
-    }
-  }
-
   String stringToString(String title) {
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/widgets/card_screen.dart]: what is this method for?

File: packages/presentation/pubspec.yaml [hunk 16]
```diff
@@ -15,12 +15,19 @@ dependencies:
 
   data:
     path: ../data
+  domain:
+    path: ../domain
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/pubspec.yaml]: no need in domain dependency here. data already has domain

File: packages/presentation/test/login_test.mocks.dart [hunk 17]
```diff
@@ -0,0 +1,74 @@
+// Mocks generated by Mockito 5.1.0 from annotations
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/test/login_test.mocks.dart]: ignore it

File: test/widget_test.dart [hunk 18]
```diff
@@ -1,29 +1,29 @@
-// This is a basic Flutter widget test.
-//
-// To perform an interaction with a widget in your test, use the WidgetTester
-// utility that Flutter provides. For example, you can send tap and scroll
-// gestures. You can also use WidgetTester to find child widgets in the widget
-// tree, read text, and verify that the values of widget properties are correct.
+// // This is a basic Flutter widget test.
```
Comment (sergei-mikhailovskii-idf) [inline test/widget_test.dart]: delete it

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

File: packages/data/lib/dio/token_interceptor.dart (hunk 1 above)
Comment (guvanch-amangeldiyev) [reply packages/data/lib/dio/token_interceptor.dart]: onResponse I am trying impl that case what you told me, what if user open the app first time and leave from the login page without log in

I want save the token onResponse if is 200
if it is 401 delete

or is it not good idea?

File: packages/presentation/lib/screen/spalsh/main_spalsh.dart (hunk 13 above)
Comment (guvanch-amangeldiyev) [reply packages/presentation/lib/screen/spalsh/main_spalsh.dart]: done

File: packages/presentation/lib/screen/home/home_page.dart (hunk 11 above)
Comment (guvanch-amangeldiyev) [reply packages/presentation/lib/screen/home/home_page.dart]: done

File: packages/presentation/lib/screen/app/my_app.dart (hunk 10 above)
Comment (guvanch-amangeldiyev) [reply packages/presentation/lib/screen/app/my_app.dart]: done

File: packages/presentation/lib/di/presentation_injector.dart (hunk 8 above)
Comment (guvanch-amangeldiyev) [reply packages/presentation/lib/di/presentation_injector.dart]: done

File: packages/domain/test/domain_repo_test.mocks.dart (hunk 7 above)
Comment (guvanch-amangeldiyev) [reply packages/domain/test/domain_repo_test.mocks.dart]: done

File: packages/domain/test/domain_repo_test.dart (hunk 6 above)
Comment (guvanch-amangeldiyev) [reply packages/domain/test/domain_repo_test.dart]: done

File: packages/data/lib/dio/dio_builder.dart (hunk 3 above)
Comment (guvanch-amangeldiyev) [reply packages/data/lib/dio/dio_builder.dart]: done

File: packages/data/lib/di/data_injector.dart (hunk 2 above)
Comment (guvanch-amangeldiyev) [reply packages/data/lib/di/data_injector.dart]: done

File: packages/presentation/lib/screen/spalsh/bloc/bloc_main.dart (hunk 12 above)
Comment (guvanch-amangeldiyev) [reply packages/presentation/lib/screen/spalsh/bloc/bloc_main.dart]: done

File: packages/presentation/lib/screen/spalsh/main_spalsh.dart (hunk 14 above)
Comment (guvanch-amangeldiyev) [reply packages/presentation/lib/screen/spalsh/main_spalsh.dart]: done

File: packages/presentation/lib/widgets/card_screen.dart (hunk 15 above)
Comment (guvanch-amangeldiyev) [reply packages/presentation/lib/widgets/card_screen.dart]: I forgot it delete, it was a mapper
I create it when we just had started

File: packages/presentation/pubspec.yaml (hunk 16 above)
Comment (guvanch-amangeldiyev) [reply packages/presentation/pubspec.yaml]: done

File: packages/presentation/test/login_test.mocks.dart (hunk 17 above)
Comment (guvanch-amangeldiyev) [reply packages/presentation/test/login_test.mocks.dart]: done

File: packages/data/test/networking_repository_mocks.mocks_test.dart (hunk 5 above)
Comment (guvanch-amangeldiyev) [reply packages/data/test/networking_repository_mocks.mocks_test.dart]: done

File: packages/data/lib/repository/network_repository.dart (hunk 4 above)
Comment (guvanch-amangeldiyev) [reply packages/data/lib/repository/network_repository.dart]: I didn't get here what you mean


here /job/*job name*/api/json

job name is url (dynamically changging)
api/json is ApiHelperCore.pathToken

File: test/widget_test.dart (hunk 18 above)
Comment (guvanch-amangeldiyev) [reply test/widget_test.dart]: done

File: packages/presentation/lib/internal/service/analitic_event.dart (hunk 9 above)
Comment (guvanch-amangeldiyev) [reply packages/presentation/lib/internal/service/analitic_event.dart]: done

File: packages/data/lib/dio/token_interceptor.dart (hunk 1 above)
Comment (sergei-mikhailovskii-idf) [reply packages/data/lib/dio/token_interceptor.dart]: but now there's nothing here

File: packages/data/lib/repository/network_repository.dart (hunk 4 above)
Comment (sergei-mikhailovskii-idf) [reply packages/data/lib/repository/network_repository.dart]: than form the path in ApiHelperCore

File: packages/presentation/lib/internal/service/analitic_event.dart (hunk 9 above)
Comment (sergei-mikhailovskii-idf) [reply packages/presentation/lib/internal/service/analitic_event.dart]: no need in loop. use replaceAll

File: packages/data/lib/dio/token_interceptor.dart [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+      var data = Map<String, dynamic>.from(response.data);
+
+      if (data["token"] != null) {
+        //!save token
+        _localStorageRepository.saveToken(data.values.toString());
+      } else if (response.statusCode == 401) {
+        _localStorageRepository.deleteToken();
+      }
+    }
+    super.onResponse(response, handler);
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/lib/dio/token_interceptor.dart]: 1) u cannot receive 401 here. 401 will go to onError
2) token is not in the response.data it is in the headers

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

**Pull request merged**